providers/types/errors.go

```diff
@@ -0,0 +1,108 @@
+package types
+
+import (
+	"errors"
+	"fmt"
+)
+
+// ErrorCode is a type alias for an int error code.
+type ErrorCode int
+
+const (
+	OK                         ErrorCode = 0
```

Comment:
chad

Reply:
nit: I would split up the errors into the directories that compose specific errors. for example i think it makes more sense for the API errors to be in the base/api directory, ws errors in the base/ws directory, etc.